src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,197 @@
+
+    /**
+     * Deletes a line from storage text file.
+     * @param line
+     */
+    private void deleteLine(String line) {
+        File tempFile = new File("temp.txt");
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(this.storageFile));
+            BufferedWriter writer = new BufferedWriter(new FileWriter(tempFile));
+
+            String rmLine = line;
+            String currLine;
+            boolean done = false;
+            boolean linesLeft = true;
+            while (linesLeft) {
+                currLine = reader.readLine();
+                if (currLine == null) {
+                    linesLeft = false;
+                } else {
+                    String trimmedLine = currLine.trim();
+                    if (trimmedLine.equals(rmLine) && !done) {
+                        done = true;
+                        continue;
+                    }
+                    writer.write(currLine + "\n");
+                }
+            }
+            writer.close();
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        tempFile.renameTo(this.storageFile);
+    }
+
+    /**
+     * Modifies a line in storage text file.
+     * @param line Line to be modified.
+     * @param newLine Line to be modified into.
+     */
+    private void modifyLine(String line, String newLine) {
+        File tempFile = new File("temp.txt");
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(this.storageFile));
+            BufferedWriter writer = new BufferedWriter(new FileWriter(tempFile));
+
+            String rmLine = line;
+            String currLine;
+            boolean done = false;
+            boolean linesLeft = true;
+            while (linesLeft) {
+                currLine = reader.readLine();
+                if (currLine == null) {
+                    linesLeft = false;
+                } else {
+                    String trimmedLine = currLine.trim();
+                    if (trimmedLine.equals(rmLine) && !done) {
+                        done = true;
+                        writer.write(newLine + "\n");
+                    } else {
+                        writer.write(currLine + "\n");
+                    }
+                }
+            }
+            writer.close();
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        tempFile.renameTo(this.storageFile);
+    }
+
+    /**
+     * Add a line of text to storage text file.
+     * @param text Line of text to be added.
+     */
+    private void appendToFile(String text) {
+        try {
+            FileWriter fw = new FileWriter(this.storageFile, true); // create a FileWriter in append mode
+            fw.write(text + "\n");
+            fw.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
```

Comment:
You can consider including the @throws section  in the JavaDocs to include situations when the method throws an Exception

Reply:
Yup thanks! I actually thought the style guide meant only header comment is necessary and not the entire javadocs